frontend/src/pages/projects/screens/spawner/storage/StorageClassSelect.tsx

```diff
@@ -0,0 +1,109 @@
+import { Split, SplitItem, Label, FormGroup, Alert, FormHelperText } from '@patternfly/react-core';
+import React from 'react';
+import SimpleSelect, { SimpleSelectOption } from '~/components/SimpleSelect';
+import useStorageClasses from '~/concepts/k8s/useStorageClasses';
+import { CreatingStorageObject, UpdateObjectAtPropAndValue } from '~/pages/projects/types';
+import { getStorageClassConfig } from '~/pages/storageClasses/utils';
+
+type StorageClassSelectProps = {
+  data: CreatingStorageObject;
+  setData: UpdateObjectAtPropAndValue<CreatingStorageObject>;
+  isEdit?: boolean;
+  menuAppendTo?: HTMLElement;
+};
+
+const StorageClassSelect: React.FC<StorageClassSelectProps> = ({
+  data,
+  setData,
+  isEdit,
+  menuAppendTo,
+}) => {
+  const [storageClasses, storageClassesLoaded] = useStorageClasses();
+
+  const enabledStorageClasses = storageClasses
+    .filter((sc) => getStorageClassConfig(sc)?.isEnabled)
+    .toSorted((a, b) => {
+      const aConfig = getStorageClassConfig(a);
+      const bConfig = getStorageClassConfig(b);
+      if (aConfig?.isDefault) {
+        return -1;
+      }
+      if (bConfig?.isDefault) {
+        return 1;
+      }
+      return (aConfig?.displayName || a.metadata.name).localeCompare(
+        bConfig?.displayName || b.metadata.name,
+      );
+    });
+
+  const isSelectedDisabled =
+    storageClasses.length <= 1 ||
+    storageClasses.some((sc) => {
+      if (sc.metadata.name === data.storageClassName) {
+        const config = getStorageClassConfig(sc);
+        return config && !config.isEnabled;
+      }
+      return false;
+    });
+  const options: SimpleSelectOption[] = enabledStorageClasses.map((sc) => {
+    const config = getStorageClassConfig(sc);
+
+    return {
+      key: sc.metadata.name,
+      label: config?.displayName || sc.metadata.name,
+      description: config?.description,
+      isDisabled: !config?.isEnabled,
+      dropdownLabel: (
+        <Split>
+          <SplitItem>{config?.displayName || sc.metadata.name}</SplitItem>
+          <SplitItem isFilled />
+          <SplitItem>
+            {config?.isDefault && (
+              <Label isCompact color="green">
+                Default class
```

Comment:
@xianli123 I'm noticing this reads `Default class` here for a Select that has a label giving context of `Storage class`, and in the storage classes table, a row with the "default" openshift label only reads `Default`. Should they not be the same?

<img width="555" alt="image" src="https://github.com/user-attachments/assets/7caa14ed-575b-4d6d-a637-deb63135aeee">

Reply:
Hi @jpuzz0.They are different. 

- In the class table, the `default` label next to the class means that this class is the default class in OpenShift.
- The `Default class` here for a 'Select' in storage creation form means that this class is the default class in RHOAI. 
Initially, RHOAI uses Openshift's default class as its default class. If the RHOAI admin chooses another enabled class as the RHOAI default class, the new RHOAI default class will be used when the new storage is created.
Is it clear?

Reply:
Yes, thanks for clarifying! Should this label have a tooltip letting the user know this is the RHOAI default and not the Openshift default?